server/src/test/java/org/elasticsearch/snapshots/ConcurrentSnapshotsIT.java

```diff
@@ -805,6 +856,14 @@ private CompletableFuture<SnapshotInfo> startAndBlockFailingFullSnapshot(String
         return future;
     }
 
+    private CompletableFuture<AcknowledgedResponse> startAndBlockOnDeleteSnapshot(String repoName, String snapshotName) throws Exception {
+        final String masterName = cluster().getMasterName();
+        mockRepo(repoName, masterName).setBlockOnDeleteIndexFile();
```

Comment:
if we have `setBlockOnAnyFiles()` then `getRepositoryData()` is already blocked and you don't have in progress deletions.

Reply:
Just fyi: I think in ES this works with the data block due to their repository data cache.

Reply:
yep, I was just trying to figure out a workaround, it affects other tests, but will try to figure out something, maybe introduce some other block type in the MockRepository.